client/src/main/java/com/linecorp/decaton/client/DecatonClient.java

```diff
@@ -79,6 +82,19 @@
      */
     CompletableFuture<PutTaskResult> put(String key, T task, Consumer<Throwable> errorCallback);
 
+    /**
+     * Put a task onto a specified kafka partition.
+     * Note that the specified partition here is prioritised over the result of
+     * {@link Partitioner#partition(String, Object, byte[], Object, byte[], Cluster)}.
+     *
+     * @param key the criteria to shuffle and order tasks. null can be specified if it doesn't matters.
+     * @param task an instance of task. Should never be null.
+     * @param partition the id of the partition
+     *
+     * @return a {@link CompletableFuture} which represents the result of task put.
+     */
+    CompletableFuture<PutTaskResult> put(String key, T task, int partition);
```

Comment:
I have a concern about this API.
If users use the following API with `int` as the argument for `timestamp`, I think the behavior will be different after this change.
https://github.com/line/decaton/blob/916403fac80bef5b222a661d836cb814f4aa9a53/client/src/main/java/com/linecorp/decaton/client/DecatonClient.java#L50

Reply:
Good point, this could be an error prone for users.

Also, if we want to specify timestamp or other task metadata with partition, we have to add other overloads and it's a mess.

So, how about adding one "universal" method instead? i.e.
```
put(String key, T task, TaskMetadata overrideTaskMetadata, Integer partition)
```
With allowing setting overrideTaskMetadata or partition to be null for users who don't need to set them explicitly.

Reply:
It seems good